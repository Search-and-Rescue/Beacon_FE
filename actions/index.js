export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const setEmergencyContacts = contacts => ({
  type: 'SET_CONTACTS',
  contacts
})

export const setVehicles = vehicles => ({
  type: 'SET_VEHICLES',
  vehicles
})

export const setGear = gear => ({
  type: 'SET_GEAR',
  gear
})

export const setTrips = trips => ({
  type: 'SET_TRIPS',
  trips
})

export const setCurrentTrip = (currentId, currentName) => ({
  type: 'SET_CURRENT',
  currentId,
  currentName
})

export const removeCurrentTrip = () => ({
  type: 'REMOVE_CURRENT'
})
