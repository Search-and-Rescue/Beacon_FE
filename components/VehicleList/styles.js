const styles = {
  vehiclesListContainer: {
    alignItems: "center",
    height: "96%",
    marginTop: 20
  },
  vehiclesList: {
    backgroundColor: "fff",
    borderColor: "#001028",
    borderRadius: 3,
    borderWidth: 4,
    flex: 1,
    height: 300,
    marginHorizontal: "5%",
    width: "90%"
  },
  backgroundImage: {
    flex: 1,
    width: "100%"
  },
  vehicleCard: {
    backgroundColor: "#efefef",
    flex: 1,
    flexDirection: "row",
    height: 50,
    marginVertical: 5,
    justifyContent: "space-between"
  },
  vehicleRemoveBtn: {
    color: "#001028",
    height: 30,
    marginHorizontal: 12,
    marginVertical: 10,
    padding: 5
  },
  vehicleName: {
    flex: 1,
    fontFamily: "Futura",
    fontSize: 21,
    paddingVertical: 10
  },
  addItemBtn: {
    borderColor: "black",
    borderWidth: 1,
    padding: 10,
    width: "100%"
  },
  addBtnText: {
    fontSize: 21,
    textAlign: "center"
  }
};

export default styles;