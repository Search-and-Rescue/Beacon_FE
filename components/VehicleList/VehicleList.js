import React, { Component } from "react";
import { View, Text, ScrollView } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import { connect } from "react-redux";
import styles from "./styles";

export class VehicleList extends Component {
  constructor(props) {
    super(props);
  }

  editVehicle = vehicle => {
    this.props.navigation.navigate("Vehicle", { vehicle });
  };

  render() {
    const vehicleLinks = this.props.vehicles.map(vehicle => {
      return (
        <TouchableOpacity 
          key={vehicle.id} 
          style={styles.vehiclesLink}
          onPress={() => this.editVehicle(vehicle)}
          >
          <Text style={styles.vehiclesName}>{vehicle.year} {vehicle.make} {vehicle.model}</Text>
        </TouchableOpacity>
      );
    });

    return (
      <View>
        <View style={styles.vehiclesListContainer}>
          <ScrollView style={styles.vehiclesList}>
            {vehicleLinks}
          </ScrollView>
          <TouchableOpacity
            style={styles.addItemBtn}
            onPress={() => this.props.navigation.navigate("Vehicle")}
          >
            <Text style={styles.addBtnText}>Add a vehicle</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}

export const mapStateToProps = ({ vehicles }) => ({
  vehicles
});

export default connect(mapStateToProps)(VehicleList);
