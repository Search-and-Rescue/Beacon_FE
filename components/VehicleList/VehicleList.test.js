import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { VehicleList, mapStateToProps, mapDispatchToProps } from './VehicleList';
import { configure } from 'enzyme';
import { setVehicles } from "../../actions";
import Adapter from 'enzyme-adapter-react-16';

jest.mock('react-native-gesture-handler', () => 'TouchableOpacity');

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: ""
}));

configure({ adapter: new Adapter() });

describe('VehicleList', () => {
  let wrapper;
  const vehiclesMock = [{
    id: 1,
    make: 'Ford',
    model: 'F150',
    year: 2010,
    color: 'silver',
    licensePlate: 'CYE 909'
  }];
  const mockState = {
    vehicles: vehiclesMock
  }

  beforeEach(() => {
    const mockVehicles = [{
      id: 1,
      make: 'Ford',
      model: 'F150',
      year: 2010,
      color: 'silver',
      licensePlate: 'CYE 909'
    }];

    wrapper = shallow(<VehicleList
      vehicles={mockVehicles}
    />);
  });

  it('should match the snapshot with all of the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should grab the props it needs', () => {
    const expected = {
      vehicles: vehiclesMock
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });

  it('it calls dispatch with the setVehicles action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setVehicles(vehiclesMock);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setVehicles(vehiclesMock);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});