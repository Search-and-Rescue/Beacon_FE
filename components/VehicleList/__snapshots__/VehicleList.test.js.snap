// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleList should match the snapshot with all of the data passed through 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "96%",
        "marginTop": 20,
      }
    }
  >
    <ScrollViewMock
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 2,
          "flex": 1,
          "height": 300,
          "width": "90%",
        }
      }
    >
      <Component
        key="1"
        style={
          Object {
            "backgroundColor": "#efefef",
            "borderColor": "black",
            "borderWidth": 1,
            "flex": 1,
            "height": 30,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 21,
              "height": 20,
            }
          }
        >
          Ford
        </Text>
      </Component>
    </ScrollViewMock>
    <Component
      onPress={[Function]}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 21,
            "textAlign": "center",
          }
        }
      >
        Add a vehicle
      </Text>
    </Component>
  </View>
</View>
`;
